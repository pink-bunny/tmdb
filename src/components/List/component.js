import React from 'react';
import PropTypes from 'prop-types';
import {
  Layout, Row, Col, Typography
} from 'antd';

import MoviesList from '../MoviesList';
import RemoveFromListModal from './RemoveFromListModal';

const ListComponent = ({
  listTitle,
  emptyListTxt,
  list,
  loading,
  error,
  totalItems,
  currentPage,
  fetchList,
  removeModalAction,
  removeModalTxt
}) => (
  <Layout>
    <Layout.Content>
      <Row>
        <Col
          offset={2}
          span={20}
        >
          <div className="top-margin">
            <Typography.Title>
              {listTitle}
            </Typography.Title>
          </div>
        </Col>
      </Row>

      <MoviesList
        list={list}
        error={error}
        loading={loading}
        onPaginationClick={fetchList}
        totalItems={totalItems}
        currentPage={currentPage}
        emptyListTxt={emptyListTxt}
        actionsList={(id, title) => ([
          <RemoveFromListModal
            key="delete"
            id={id}
            title={title}
            removeModalAction={removeModalAction}
            removeModalTxt={removeModalTxt}
          />
        ])}
      />

    </Layout.Content>
  </Layout>
);

ListComponent.propTypes = {
  listTitle: PropTypes.string.isRequired,
  emptyListTxt: PropTypes.string.isRequired,
  fetchList: PropTypes.func.isRequired,
  removeModalAction: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string.isRequired,
  removeModalTxt: PropTypes.string.isRequired,
  list: PropTypes.arrayOf(PropTypes.object),
  totalItems: PropTypes.number.isRequired,
  currentPage: PropTypes.number.isRequired
};
ListComponent.defaultProps = {
  list: null
};

export default ListComponent;
