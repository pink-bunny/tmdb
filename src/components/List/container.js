import React from 'react';
import PropTypes from 'prop-types';

import ListComponent from './component';

class ListContainer extends React.Component {
  componentDidMount() {
    const { fetchList } = this.props;
    fetchList();
  }

  render() {
    const {
      listTitle,
      emptyListTxt,
      list,
      loading,
      error,
      totalItems,
      currentPage,
      fetchList,
      removeModalAction,
      removeModalTxt
    } = this.props;

    return (
      <ListComponent
        listTitle={listTitle}
        emptyListTxt={emptyListTxt}
        list={list}
        loading={loading}
        error={error}
        totalItems={totalItems}
        currentPage={currentPage}
        fetchList={fetchList}
        removeModalAction={removeModalAction}
        removeModalTxt={removeModalTxt}
      />
    );
  }
}

ListContainer.propTypes = {
  removeModalTxt: PropTypes.string.isRequired,
  listTitle: PropTypes.string.isRequired,
  emptyListTxt: PropTypes.string.isRequired,
  fetchList: PropTypes.func.isRequired,
  removeModalAction: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string.isRequired,
  list: PropTypes.arrayOf(PropTypes.object),
  totalItems: PropTypes.number.isRequired,
  currentPage: PropTypes.number.isRequired
};
ListContainer.defaultProps = {
  list: null
};

export default ListContainer;
