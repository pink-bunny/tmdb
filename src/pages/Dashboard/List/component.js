import React from 'react';
import PropTypes from 'prop-types';
import {
  Row,
  Col,
  Pagination,
  Empty,
  Typography,
  Spin
} from 'antd';

import Movie from './MovieItem';

const List = ({
  list,
  loading,
  error,
  totalItems,
  fetchMovies,
  currentPage
}) => {
  if (loading) {
    return (
      <Row type="flex" justify="center">
        <Col>
          <Spin />
        </Col>
      </Row>
    );
  }

  if (error) {
    return (
      <Row type="flex" justify="center">
        <Col>
          <Typography.Paragraph type="danger">
            {error}
          </Typography.Paragraph>
        </Col>
      </Row>
    );
  }

  if (!list) {
    return (
      <Empty
        description="No movies found"
        image={Empty.PRESENTED_IMAGE_SIMPLE}
      />
    );
  }

  return (
    <>
      <Row
        type="flex"
        gutter={16}
      >
        <Col
          span={20}
          offset={2}
        >
          {list.map((item) => (
            <Col
              key={item.id}
              xs={{ span: 12 }}
              sm={{ span: 12 }}
              md={{ span: 8 }}
              lg={{ span: 6 }}
              xl={{ span: 4 }}
            >
              <Movie
                title={item.title}
                overview={item.overview}
                poster={item.poster_path}
              />
            </Col>
          ))}
        </Col>
      </Row>
      {totalItems > 20 ? (
        <Row
          type="flex"
          justify="center"
        >
          <Col>
            <Pagination
              current={currentPage}
              pageSize={20}
              total={totalItems}
              className="pagination"
              onChange={fetchMovies}
            />
          </Col>
        </Row>
      ) : (
        null
      )}
    </>
  );
};

List.propTypes = {
  list: PropTypes.arrayOf(PropTypes.object).isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string.isRequired,
  totalItems: PropTypes.number.isRequired,
  fetchMovies: PropTypes.func.isRequired,
  currentPage: PropTypes.number.isRequired
};

export default List;
