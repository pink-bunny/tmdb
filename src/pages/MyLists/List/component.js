import React from 'react';
import PropTypes from 'prop-types';
import {
  Row, Col, Card, Typography, Icon, Pagination, Spin, Empty
} from 'antd';

const List = ({
  list,
  loading,
  error
}) => {
  if (loading) {
    return (
      <Row type="flex" justify="center">
        <Col>
          <Spin />
        </Col>
      </Row>
    );
  }

  if (error) {
    return (
      <Row type="flex" justify="center">
        <Col>
          <Typography.Paragraph type="danger">
            {error}
          </Typography.Paragraph>
        </Col>
      </Row>
    );
  }

  if (list.length === 0) {
    return (
      <Empty description="No results" />
    );
  }

  return (
    <>
      <Row
        gutter={8}
        type="flex"
      >
        <Col
          span={20}
          offset={2}
        >
          {list.map((item) => (
            <Col
              key={item.id}
              xs={{ span: 12 }}
              sm={{ span: 8 }}
              md={{ span: 6 }}
              lg={{ span: 4 }}
              xl={{ span: 4 }}
            >
              <Card
                hoverable
                className="top-margin"
                actions={[<Icon
                  key="delete"
                  type="delete"
                />]}
              >
                <Typography.Title level={4}>
                  {item.name}
                </Typography.Title>
                {item.description}
              </Card>
            </Col>
          ))}
        </Col>
      </Row>
    </>
  );
};

List.propTypes = {
  list: PropTypes.arrayOf(PropTypes.object).isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string.isRequired
};

export default List;
