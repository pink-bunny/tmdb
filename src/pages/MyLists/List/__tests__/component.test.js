import React from 'react';
import { shallow } from 'enzyme';

import List from '../component';

describe('List component', () => {
  const requiredProps = {
    loading: false,
    error: ''
  };

  it('matches snapshot when it is loading', () => {
    const props = {
      ...requiredProps,
      loading: true
    };
    const wrapper = shallow(<List {...props} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot when it has error', () => {
    const props = {
      ...requiredProps,
      error: 'Loading error'
    };
    const wrapper = shallow(<List {...props} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot with empty list', () => {
    const wrapper = shallow(<List {...requiredProps} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot with list', () => {
    const props = {
      ...requiredProps,
      list: [{
        id: 1,
        name: 'My favorites movies',
        description: 'Description of my favorites movies'
      }]
    };
    const wrapper = shallow(<List {...props} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });
});
