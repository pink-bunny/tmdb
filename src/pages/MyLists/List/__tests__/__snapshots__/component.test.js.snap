// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component matches snapshot when it has error 1`] = `
<Row
  gutter={0}
  justify="center"
  type="flex"
>
  <Col>
    <Paragraph
      type="danger"
    >
      Loading error
    </Paragraph>
  </Col>
</Row>
`;

exports[`List component matches snapshot when it is loading 1`] = `
<Row
  gutter={0}
  justify="center"
  type="flex"
>
  <Col>
    <Spin
      size="default"
      spinning={true}
      wrapperClassName=""
    />
  </Col>
</Row>
`;

exports[`List component matches snapshot with empty list 1`] = `
<Empty
  description="No results"
/>
`;

exports[`List component matches snapshot with list 1`] = `
<Fragment>
  <Row
    gutter={8}
    type="flex"
  >
    <Col
      offset={2}
      span={20}
    >
      <Col
        key="1"
        lg={
          Object {
            "span": 4,
          }
        }
        md={
          Object {
            "span": 6,
          }
        }
        sm={
          Object {
            "span": 8,
          }
        }
        xl={
          Object {
            "span": 4,
          }
        }
        xs={
          Object {
            "span": 12,
          }
        }
      >
        <Card
          actions={
            Array [
              <Memo(Connect(WrappedComponentShell))
                id={1}
                name="My favorites movies"
              />,
            ]
          }
          className="top-margin"
        >
          <Title
            level={4}
          >
            <Link
              to="/my-lists/1"
            >
              My favorites movies
            </Link>
          </Title>
          Description of my favorites movies
        </Card>
      </Col>
    </Col>
  </Row>
</Fragment>
`;
